pkg/metrics/metrics.go

```diff
@@ -143,11 +146,37 @@ func (metrics *metrics) ObserveOngoingStateDuration(run *api.PipelineRun) error
 	return nil
 }
 
-func (metrics *metrics) ObserveUpdateDurationByType(typ string, duration time.Duration) {
-	metrics.Update.With(prometheus.Labels{"type": typ}).Observe(duration.Seconds())
+// ObserveDuration logs the duration of updates by type
+func (metrics *metrics) ObserveDuration(duration time.Duration, retry bool) {
+	caller := callerFunctionName(1, 1)
```

Comment:
`callerFunctionName` works upon the stack frames. In case someone refactors the coding later and introduces new functions "in between" this will affect the reporting and maybe no-one will understand why the reporting suddenly looks different. Would that be an issue?